services/QuillLMS/app/workers/snapshots/cache_snapshot_top_x_worker.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Snapshots
+  class CacheSnapshotTopXWorker
+    include Sidekiq::Worker
+
+    PUSHER_EVENT = 'admin-snapshot-top-x-cached'
+
+    QUERIES = {
+      'top-concepts-assigned' => Snapshots::TopConceptsAssignedQuery,
+      'top-concepts-practiced' => Snapshots::TopConceptsPracticedQuery,
+      'most-active-schools' => Snapshots::MostActiveSchoolsQuery,
+    }
+
+    def perform(cache_key, query, user_id, timeframe, school_ids, grades)
+      payload = generate_payload(query, timeframe, school_ids, grades)
+
+      Rails.cache.write(cache_key, payload, expires_in: timeframe[:current_end] + 1.day)
+
+      PusherTrigger.run(user_id, PUSHER_EVENT,
+        {
+          query: query,
+          timeframe: timeframe[:name],
+          school_ids: school_ids,
+          grades: grades
+        }
+      )
+    end
+
+    private def generate_payload(query, timeframe, school_ids, grades)
+      timeframe_start = timeframe[:current_start]
```

Comment:
not sure you gain much from assigning these local vars over just passing `timeframe[:current_start]` directly into the method below.